Empire/Empire.cpp

```diff
@@ -166,6 +167,11 @@ namespace {
                 elem.allocated_rp = 0.0f;
             }
         }
+
+        if (total_RPs_spent >= RPs - EPSILON)
+            return; // don't further allocate if nothing left to spend
```

Comment:
This early exit code is now pointless. It's left over from when I had the auto-allocation in SetTechQueueElementSpending, but it's been moved to CheckResearchProgress.